src/tls_tpm.c

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2020 Xaptum, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "tls_tpm.h"
+#include "log.h"
+#include "tls.h"
+#include <openssl/engine.h>
+#include <openssl/ssl.h>
+#include <openssl/ui.h>
+#include <stdio.h>
+#include <string.h>
+
+/* Forward declarations */
+static UI_METHOD*
+enftun_tls_tpm_ssl_ui_passthrough(void);
+
+static ENGINE*
+enftun_tls_tpm_engine_init()
+{
+    ENGINE* eng                = NULL;
+    const char* engine_names[] = {"tpm2tss", "tpm2"};
+    long unsigned int engine_index;
+
+    /* Init the engine */
+    ENGINE_load_builtin_engines();
+
+    engine_index = 0;
+    for (engine_index = 0;
+         engine_index < sizeof(engine_names) / sizeof(*engine_names);
```

Comment:
`array[0]` is more conanonical that `*array`, I think:
```
engine_index < sizeof(engine_names) / sizeof(engine_names[0])`
```

Reply:
Changed